src/parsers/locale-messagesjson.js

```diff
@@ -153,6 +166,23 @@ export default class LocaleMessagesJSONParser extends JSONParser {
         );
       }
 
+      if (messageNameMatch) {
+        const messageName = messageNameMatch[1];
+        let nameContainsInvalidWords = false;
+        if (message === messageName) {
```

Comment:
If you make this `if` the toplevel one, we avoid some unnecessary work when we are not looking at `name`.